### PR TITLE
rgw/archive: avoid duplicating objects when syncing from multiple zones

### PR DESCRIPTION
**Pull request by smanjara**
Fixes https://tracker.ceph.com/issues/59184

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: src/rgw/driver/rados/rgw_cr_rados.cc [hunk 1]
```diff
@@ -740,15 +740,23 @@ int RGWAsyncFetchRemoteObj::_send_request(const DoutPrefixProvider *dpp)
...
   rgw::sal::RadosObject dest_obj(store, dest_key.value_or(key), &dest_bucket);
-    
+
+  rgw_obj stat_dest_obj;
+
+  if (!stat_follow_olh) {
+    stat_dest_obj = dest_obj.get_obj();
+  } else {
+    stat_dest_obj = src_obj;
+  }
```
[the remark this answers is not in the document]
Comment (mattbenjamin) [reply src/rgw/driver/rados/rgw_cr_rados.cc:750]: thank you for the detailed explanation, @cbodley

[the remark this answers is not in the document]
Comment (smanjara) [reply src/rgw/driver/rados/rgw_cr_rados.cc:750]: ok yes I overlooked the bucket replication policy case. Your proposal makes sense. will open a tracker and follow up with a pr soon.

**Comment by cbodley** [comment]
@smanjara did you have further changes that resolved the bucket marker issue? i don't see them here

**Comment by smanjara** [comment]
> 

Pushed an update. I guess I should have made a separate commit to tell the difference but this is the line that changed:
https://github.com/ceph/ceph/pull/50841/commits/7085791a7fcd845a7cc12dc32f8499225372b969#diff-7481fe6c19efcbe41be74e93e71fbc1e83646849199c979a936faf2d83d5a65fR2933

**Review by cbodley — APPROVED** [review]

**Comment by cbodley** [comment]
passed qa after rerun https://pulpito.ceph.com/cbodley-2023-05-25_13:09:32-rgw-wip-cbodley-testing-distro-default-smithi/

**Pull request merged**